packages/components/src/utils/validation/validateProps.ts

```diff
@@ -241,3 +256,17 @@ export const validateProps = <T extends Class<any>>(instance: InstanceType<T>, p
       printErrorMessage({ ...error, componentName: getTagNameWithoutPrefix(instance.host as HTMLElement) })
     );
 };
+
+/**
+ * Check if all elements in an array have types that are allowed.
+ * @template T - A union of allowed types for the array elements.
+ * @param {any} arr - The array to be validated.
+ * @param {T[]} allowedTypes - An array of allowed types.
+ * @returns {boolean} - True if all elements have allowed types, false otherwise.
+ */
+export const isValidArray = <T extends AllowedArrayTypes>(arr: any, allowedTypes: T[]): boolean => {
+  if (!Array.isArray(arr)) {
+    return false;
+  }
+  return arr.every((item) => allowedTypes.includes(typeof item as T));
```

Comment:
Isn't there a `typeof` missing or why not reuse our `number` and `string` validators?